cmd/doctor.go

```diff
@@ -495,3 +501,82 @@ func runDoctorScriptType(ctx *cli.Context) ([]string, error) {
 	}
 	return []string{fmt.Sprintf("ScriptType %s is on the current PATH at %s", setting.ScriptType, path)}, nil
 }
+
+func runDoctorCheckDBConsistency(ctx *cli.Context) ([]string, error) {
+	var results []string
+	var outdatedDB bool
+
+	// make sure DB version is uptodate
+	if err := models.NewEngine(context.Background(), migrations.EnsureUpToDate); err != nil {
+		results = append(results, "Warning: model version on the database does not match the current Gitea version. Model consistency can be checked but not fixed until the database is upgraded.")
+		outdatedDB = true
```

Comment:
If the DB is out-of-date and it is not automatically fixed all further checking must stop.

Reply:
We can do it the one way or the oter one BUT pleace we ned a dessission!!

Reply:
Sorry, @6543 I was being accomodating about the tests; for me _fixing_ when the version is incorrect is a no-no, and I think that checking is harmless (with a warning). Although @zeripath has a point, my reasoning behind allowing the tests even if the version is outdated is because the inconsistencies could make a migration fail, so the user is stuck between a fix they cannot run and a migration that fails and they doesn't know why. That's why I thought we could give the user (and ourselves, when we provide support) a hint on why the migration could be failing and so they can resort to a previous version with the doctor to apply the fix.

Reply:
It turns out that invoking `models.NewEngine()` is a problem because it will call `.Sync2()` on tables in certain circumstances. That's a problem because even for what should be a read-only operation (checking) we could be inadvertently upgrading the model. For example here:

https://github.com/go-gitea/gitea/blob/e25348da889718f2116f23e25aacb10ca8aee2a3/models/models.go#L185-L205

Maybe we could set up a new function specifically for the doctor that never calls any `Sync2()` operation, or add a flag to the existing function.

Reply:
@guillep2k then we have a problem, **runDoctorCheckDBVersion** already use it in this way

Reply:
Sync2 will only be run here if the version is correct.

Reply:
> 
> 
> Sync2 will only be run here if the version is correct.

I agree... the flag is not needed. I meant we need to make sure it stays like that; e.g. make a separate function or at least an explicit comment as for why sync2 must not be invoked. My concern is that someone in the future could modify this function without realizing the problem it would cause in the doctor.